File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/forecast/src/main/java/com/example/forecast/DeleteDataset.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.forecast;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.forecast.ForecastClient;
import software.amazon.awssdk.services.forecast.model.DeleteDatasetRequest;
import software.amazon.awssdk.services.forecast.model.ForecastException;


/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class DeleteDataset {

    public static void main(String[] args) {

        final String USAGE = "\n" +
                "Usage:\n" +
                "    <datasetARN> \n\n" +
                "Where:\n" +
                "    datasetARN - the ARN of the data set to delete. \n\n" ;

         if (args.length != 1) {
              System.out.println(USAGE);
              System.exit(1);
           }

        String datasetARN = args[0];
        Region region = Region.US_WEST_2;
        ForecastClient forecast = ForecastClient.builder()
                .region(region)
                .build();

        deleteForecastDataSet(forecast, datasetARN);
        forecast.close();
    }

    public static void deleteForecastDataSet(ForecastClient forecast, String myDataSetARN) {

        try {
            DeleteDatasetRequest deleteRequest = DeleteDatasetRequest.builder()
                    .datasetArn(myDataSetARN)
                    .build();

            forecast.deleteDataset(deleteRequest);
            System.out.println("The Data Set was deleted") ;

        } catch (ForecastException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/forecast/src/main/kotlin/com/kotlin/forecast/DeleteDataset.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.forecast

import aws.sdk.kotlin.services.forecast.ForecastClient
import aws.sdk.kotlin.services.forecast.model.DeleteDatasetRequest
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args:Array<String>) {

    val usage = """
    Usage:
        <dataSetARN>  

    Where:
        dataSetARN - the ARN of the data set to delete. 
       """

      if (args.size != 1) {
          println(usage)
          exitProcess(0)
      }

    val dataSetARN = args[0]
    deleteForecastDataSet(dataSetARN)
    }

suspend fun deleteForecastDataSet(myDataSetARN: String?) {

       val request = DeleteDatasetRequest {
        datasetArn = myDataSetARN
        }

        ForecastClient { region = "us-west-2" }.use { forecast ->
            forecast.deleteDataset(request)
            println("$myDataSetARN data set was deleted")
    }
}
